tools/go/def.bzl

```diff
@@ -1,6 +1,51 @@
-load("@io_bazel_rules_go//go:def.bzl", _go_library = "go_library")
+load("@io_bazel_rules_go//go/private/rules:library.bzl", _go_library = "go_library")
+load("@io_bazel_rules_go//go/private/rules:test.bzl", "go_test_kwargs")
 load("@bazel_gazelle//:deps.bzl", _go_repository = "go_repository")
 
+def _go_test_transition_impl(settings, attr):
+    """Edge transition to add minimal or mainnet build tags"""
+    settings = dict(settings)
+
+    if attr.eth_network == "minimal":
+        settings["//proto:network"] = "minimal"
+        settings["@io_bazel_rules_go//go/config:tags"] += ["minimal"]
+    elif attr.eth_network == "mainnet": # Default / optional
+        settings["//proto:network"] = "mainnet"
+        settings["@io_bazel_rules_go//go/config:tags"] += ["mainnet"]
+
+    if attr.gotags:
+        settings["@io_bazel_rules_go//go/config:tags"] += attr.gotags
+
+    return settings
+
+go_test_transition = transition(
+    implementation = _go_test_transition_impl,
+    inputs = [
+        "@io_bazel_rules_go//go/config:tags",
+        "//proto:network",
+    ],
+    outputs = [
+        "@io_bazel_rules_go//go/config:tags",
+        "//proto:network",
+    ],
+)
+
+def _go_test_transition_rule(**kwargs):
+    """A wrapper around go_test to add an eth_network attribute and incoming edge transition to support compile time configuration"""
+    kwargs = dict(kwargs)
+    attrs = dict(kwargs["attrs"])
+    attrs.update({
+        "eth_network": attr.string(values = ["mainnet", "minimal"]),
+        "_whitelist_function_transition": attr.label(
+            default = "@bazel_tools//tools/whitelists/function_transition_whitelist",
+        ),
+    })
+    kwargs["attrs"] = attrs
+    kwargs["cfg"] = go_test_transition
+    return rule(**kwargs)
+
+go_test = _go_test_transition_rule(**go_test_kwargs)
```

Comment:
This is the meat of the PR logic. 

On line 47, we define a new `go_test` which uses the same kwargs as go_test with some additions.